File: src/main/java/com/thehecklers/kcupservice/KcupServiceApplication.java
package com.thehecklers.kcupservice;

import org.springframework.boot.CommandLineRunner;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.Bean;

import java.util.Arrays;

@SpringBootApplication
class KcupServiceApplication {
	@Bean
	CommandLineRunner demoData(CoffeeRepository repository) {
		return args -> {
			repository.deleteAll();

			Arrays.asList("Espresso Roast", "Kaldi Coffee", "SK Coffee", "Klatch Dark Roast")
					.stream()
					.map(Coffee::new)
					.forEach(repository::save);

			repository.findAll().forEach(System.out::println);
		};
	}
	public static void main(String[] args) {
		SpringApplication.run(KcupServiceApplication.class, args);
	}
}

File: src/main/kotlin/com/thehecklers/kcupservice/KcupServiceApplication.kt
package com.thehecklers.kcupservice

import org.springframework.boot.CommandLineRunner
import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.context.annotation.Bean
import java.util.*
import java.util.function.Consumer

@SpringBootApplication
internal class KcupServiceApplication {
    @Bean
    fun demoData(repository: CoffeeRepository) = CommandLineRunner {
        repository.deleteAll()

        listOf("Espresso Roast", "Kaldi Coffee", "SK Coffee", "Klatch Dark Roast")
            .map { Coffee(type = it) }
            .forEach { repository.save(it) }

        repository.findAll().forEach { println(it) }
    }
}

fun main(args: Array<String>) {
    runApplication<KcupServiceApplication>(*args)
}

